ggml/src/ggml-cpu/ggml-cpu-aarch64.cpp

```diff
@@ -279,12 +316,12 @@ static void quantize_q8_0_4x4(const float * restrict x, void * restrict vy, int6
 #endif
 }
 
-static void quantize_q8_0_4x8(const float * restrict x, void * restrict vy, int64_t k) {
+static void quantize_q8_0_4x8(const float * x, void * vy, int64_t k) {
```

Comment:
I wonder if removing `restrict` here may have a negative effect on performance. Maybe it could be kept for the compilers that support it, e.g. gcc and clang have `__restrict` in C++.

Reply:
I don't know for the performance effect.

I can try to add macro like with AVX:
https://github.com/ggerganov/llama.cpp/blob/59f4db10883a4f3e855cffbf2c3ab68430e95272/ggml/src/ggml-cpu/amx/mmq.cpp#L25-L29

Well may be more like that:
```c++
#if defined(__GNUC__)
#define RESTRICT __restrict__
#elif  defined(__clang__)
#define RESTRICT __restrict
[...]
#else
#define RESTRICT
#endif
```
or even
```c++
#if defined(__GNUC__)
#define restrict __restrict__
#elif  defined(__clang__)
#define restrict __restrict
[...]
#else
#define restrict
#endif
```
(the last will make less change in this PR and may be less when/if c++ allow this keyword

Reply:
Maybe `GGML_RESTRICT` and just update the definition in `ggml.h` to support C++.